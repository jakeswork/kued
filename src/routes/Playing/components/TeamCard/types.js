import {
  shape,
  string,
  arrayOf,
  number,
} from 'prop-types';

export const propTypes = {
  classes: shape({}).isRequired,
  team: shape({
    lastPlayed: shape({
      id: string,
      opponent: string,
      outcome: string,
      rating: string,
      time: string,
    }),
    name: string,
    players: arrayOf(
      shape({
        charname: string,
        damageDone: string,
        deaths: string,
        healingDone: string,
        killingBlows: string,
        matchmaking_change: string,
        personal_change: string,
        realm: string,
        teamname: string,
        teamnamerich: string,
      }),
    ),
    rank: number,
    realm: string,
    url: string,
  }),
};

export const defaultProps = {
  team: {
    lastPlayed: {
      id: '',
      opponent: '',
      outcome: '',
      rating: '',
      time: '',
    },
    name: '',
    players: [
      {
        charname: '',
        damageDone: '',
        deaths: '',
        healingDone: '',
        killingBlows: '',
        matchmaking_change: '',
        personal_change: '',
        realm: '',
        teamname: '',
        teamnamerich: '',
      },
    ],
    rank: null,
    realm: '',
    url: '',
  },
};
