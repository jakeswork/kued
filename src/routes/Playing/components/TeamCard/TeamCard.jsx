import React, { useState } from 'react';
import { UnmountClosed as Collapse } from 'react-collapse';
import { presets } from 'react-motion';
import { FiChevronDown, FiChevronUp } from 'react-icons/fi';
import classNames from 'classnames';

import { propTypes, defaultProps } from './types';
import Card from '../../../../components/Card';
import Text from '../../../../components/Text';
import Players from './components/Players';

const TeamCard = ({ classes, team }) => {
  const [opened, setOpened] = useState(false);

  return (
    <Card className={classes.card} onClick={() => setOpened(!opened)} data-test-id="Card">
      <span className={classes.outcome}>
        <Text>
          { `${team.lastPlayed.outcome === 'Loss' ? 'Lost' : 'Won'} their last game` }
        </Text>
      </span>
      <div className={classes.root}>
        <div className={classes.small}>
          <div>
            <Text caption>
              Rank
            </Text>
            <Text bold>
              { team.rank }
            </Text>
          </div>
          <div>
            <Text caption>
              Rating
            </Text>
            <Text bold>
              { team.lastPlayed.rating }
            </Text>
          </div>
        </div>
        <div className={classes.nameWrapper}>
          <Text h4>
            { team.name }
          </Text>
          <Text caption bold>
            { team.realm }
          </Text>
        </div>
        <div className={classNames(classes.small, classes.lastPlayed)}>
          <Text caption>
            Last Played
          </Text>
          <Text bold>
            { team.lastPlayed.time }
          </Text>
        </div>
      </div>
      <Collapse
        isOpened={opened}
        className={classes.collapse}
        springConfig={presets.gentle}
      >
        <Players players={team.players} />
      </Collapse>
      {
        opened
          ? <FiChevronUp className={classes.icon} data-test-id="ChevronUp" />
          : <FiChevronDown className={classes.icon} data-test-id="ChevronDown" />
      }
    </Card>
  );
};

TeamCard.propTypes = propTypes;

TeamCard.defaultProps = defaultProps;

export default TeamCard;
