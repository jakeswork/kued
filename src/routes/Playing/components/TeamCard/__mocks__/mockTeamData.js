export default {
  lastPlayed: {
    id: '',
    opponent: '',
    outcome: 'Victory',
    rating: '',
    time: '1 hour ago',
  },
  name: 'test',
  rank: null,
  realm: '',
  url: '',
  players: [{
    charname: '',
    damageDone: '',
    deaths: '',
    healingDone: '',
    killingBlows: '',
    matchmaking_change: '',
    personal_change: '',
    realm: '',
    teamname: '',
    teamnamerich: '',
  }],
};
