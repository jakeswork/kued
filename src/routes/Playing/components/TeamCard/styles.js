export default {
  card: {
    width: 'calc(100% - 64px)',
    margin: '16px 0',
    cursor: 'pointer',
  },
  root: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  icon: {
    fontSize: 24,
    marginTop: 16,
  },
  small: {
    width: '20%',
    display: 'flex',
    justifyContent: 'space-between',
  },
  lastPlayed: {
    flexDirection: 'column',
  },
};
