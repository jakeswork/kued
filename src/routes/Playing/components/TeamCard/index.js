import withStyles from 'react-jss';

import TeamCard from './TeamCard';
import styles from './styles';

export default withStyles(styles)(TeamCard);
