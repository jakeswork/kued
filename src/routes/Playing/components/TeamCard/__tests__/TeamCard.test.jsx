import React from 'react';
import { shallow } from 'enzyme';

import TeamCard from '../TeamCard';
import mockTeamData from '../__mocks__/mockTeamData';

const defaultProps = {
  classes: {},
  team: mockTeamData,
};

describe('The TeamCard instance', () => {
  describe('When rendered with default props', () => {
    it('should render without throwing an error', () => {
      const wrapper = shallow(<TeamCard {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When clicking the component', () => {
    it('Should render the opposite chevron', () => {
      const wrapper = shallow(<TeamCard {...defaultProps} />);
      const Card = wrapper.find({ 'data-test-id': 'Card' });
      const ChevronUp = wrapper.find({ 'data-test-id': 'ChevronUp' });
      const ChevronDown = wrapper.find({ 'data-test-id': 'ChevronDown' });

      expect(ChevronDown.exists()).toBe(true);
      expect(ChevronUp.exists()).toBe(false);

      Card.simulate('click');
    });
  });

  describe('When a team has a loss', () => {
    it('Should render loss text', () => {
      const newProps = {
        ...defaultProps,
        team: {
          ...defaultProps.team,
          lastPlayed: {
            ...defaultProps.team.lastPlayed,
            outcome: 'Loss',
          },
        },
      };
      const wrapper = shallow(<TeamCard {...newProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
