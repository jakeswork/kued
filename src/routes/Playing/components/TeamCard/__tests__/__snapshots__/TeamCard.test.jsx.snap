// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TeamCard instance When a team has a loss Should render loss text 1`] = `
<Jss(Card)
  className={null}
  data-test-id="Card"
  onClick={[Function]}
>
  <span>
    <Jss(Text)
      analyticsLabel={null}
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Lost their last game
    </Jss(Text)>
  </span>
  <div>
    <div>
      <div>
        <Jss(Text)
          analyticsLabel={null}
          bold={false}
          caption={true}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        >
          Rank
        </Jss(Text)>
        <Jss(Text)
          analyticsLabel={null}
          bold={true}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        />
      </div>
      <div>
        <Jss(Text)
          analyticsLabel={null}
          bold={false}
          caption={true}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        >
          Rating
        </Jss(Text)>
        <Jss(Text)
          analyticsLabel={null}
          bold={true}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        />
      </div>
    </div>
    <div>
      <Jss(Text)
        analyticsLabel={null}
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        link={null}
        style={Object {}}
      >
        test
      </Jss(Text)>
      <Jss(Text)
        analyticsLabel={null}
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      />
    </div>
    <div
      className=""
    >
      <Jss(Text)
        analyticsLabel={null}
        bold={false}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        Last Played
      </Jss(Text)>
      <Jss(Text)
        analyticsLabel={null}
        bold={true}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        1 hour ago
      </Jss(Text)>
    </div>
  </div>
  <UnmountClosed
    isOpened={false}
    springConfig={
      Object {
        "damping": 14,
        "stiffness": 120,
      }
    }
  >
    <Jss(Players)
      players={
        Array [
          Object {
            "charname": "",
            "damageDone": "",
            "deaths": "",
            "healingDone": "",
            "killingBlows": "",
            "matchmaking_change": "",
            "personal_change": "",
            "realm": "",
            "teamname": "",
            "teamnamerich": "",
          },
        ]
      }
    />
  </UnmountClosed>
  <FiChevronDown
    data-test-id="ChevronDown"
  />
</Jss(Card)>
`;

exports[`The TeamCard instance When rendered with default props should render without throwing an error 1`] = `
<Jss(Card)
  className={null}
  data-test-id="Card"
  onClick={[Function]}
>
  <span>
    <Jss(Text)
      analyticsLabel={null}
      bold={false}
      caption={false}
      h1={false}
      h2={false}
      h3={false}
      h4={false}
      h5={false}
      link={null}
      style={Object {}}
    >
      Won their last game
    </Jss(Text)>
  </span>
  <div>
    <div>
      <div>
        <Jss(Text)
          analyticsLabel={null}
          bold={false}
          caption={true}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        >
          Rank
        </Jss(Text)>
        <Jss(Text)
          analyticsLabel={null}
          bold={true}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        />
      </div>
      <div>
        <Jss(Text)
          analyticsLabel={null}
          bold={false}
          caption={true}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        >
          Rating
        </Jss(Text)>
        <Jss(Text)
          analyticsLabel={null}
          bold={true}
          caption={false}
          h1={false}
          h2={false}
          h3={false}
          h4={false}
          h5={false}
          link={null}
          style={Object {}}
        />
      </div>
    </div>
    <div>
      <Jss(Text)
        analyticsLabel={null}
        bold={false}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={true}
        h5={false}
        link={null}
        style={Object {}}
      >
        test
      </Jss(Text)>
      <Jss(Text)
        analyticsLabel={null}
        bold={true}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      />
    </div>
    <div
      className=""
    >
      <Jss(Text)
        analyticsLabel={null}
        bold={false}
        caption={true}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        Last Played
      </Jss(Text)>
      <Jss(Text)
        analyticsLabel={null}
        bold={true}
        caption={false}
        h1={false}
        h2={false}
        h3={false}
        h4={false}
        h5={false}
        link={null}
        style={Object {}}
      >
        1 hour ago
      </Jss(Text)>
    </div>
  </div>
  <UnmountClosed
    isOpened={false}
    springConfig={
      Object {
        "damping": 14,
        "stiffness": 120,
      }
    }
  >
    <Jss(Players)
      players={
        Array [
          Object {
            "charname": "",
            "damageDone": "",
            "deaths": "",
            "healingDone": "",
            "killingBlows": "",
            "matchmaking_change": "",
            "personal_change": "",
            "realm": "",
            "teamname": "",
            "teamnamerich": "",
          },
        ]
      }
    />
  </UnmountClosed>
  <FiChevronDown
    data-test-id="ChevronDown"
  />
</Jss(Card)>
`;
